src/uct/ib/base/ib_md.c

```diff
@@ -874,19 +885,23 @@ uct_ib_mem_advise(uct_md_h uct_md, uct_mem_h memh, void *addr,
     return UCS_OK;
 }
 
-static ucs_status_t uct_ib_mkey_pack(uct_md_h uct_md, uct_mem_h uct_memh,
-                                     void *rkey_buffer)
+static ucs_status_t
+uct_ib_mkey_pack(uct_md_h uct_md, uct_mem_h uct_memh,
+                 const uct_md_mkey_pack_params_t *params,
+                 void *rkey_buffer)
 {
-    uct_ib_md_t *md         = ucs_derived_of(uct_md, uct_ib_md_t);
-    uct_ib_mem_t *memh      = uct_memh;
+    uct_ib_md_t *md     = ucs_derived_of(uct_md, uct_ib_md_t);
+    uct_ib_mem_t *memh  = uct_memh;
+    unsigned flags      = uct_md_mkey_pack_param_flags(params);
+    uint32_t rkey       = memh->rkey;
+    ucs_status_t status = UCS_OK;
```

Comment:
maybe not init here, but later for example in line 936 (else) - as later as possible